internal/validator/handler.go

```diff
@@ -86,6 +129,37 @@ func (this *handler) handleRequest(
 		return
 	}
 
+	this.vChan <- messageInfo{Request: request, Data: data, Response: ingressResponse}
+}
+
+func (this *handler) validationProcess(
+	ctx context.Context,
+	kafkaTimeout int,
+	vWg *sync.WaitGroup,
+) {
+	defer vWg.Done()
+	defer utils.GetLogFromContext(ctx).Debug("Validator stopped")
+	defer this.producer.Close()
```

Comment:
It's cleaner if the same component that opens a resource (e.g. opens a file, creates kafka producer, etc.) is the one who also closes it. I.e. if Start() in main.go does the opening then the cleanup should be somewhere close. This function should be focused on initializing the validation worker (and closing any resources opened in the process), not closing a producer that somebody else started.